srml/balances/src/lib.rs

```diff
@@ -359,6 +367,35 @@ impl<T: Trait> Module<T> {
 			<TotalIssuance<T>>::put(v);
 		}
 	}
+
+	/// Returns `Ok` iff the account is able to make a withdrawal of the given amount
+	/// for the given reason.
+	/// 
+	/// `Err(...)` with the reason why not otherwise.
+	pub fn ensure_account_can_withdraw(
```

Comment:
Should this be moved to `Currency` trait or `LockableCurrency` trait?

Reply:
No, because it is used within the module (`pub fn make_transfer`) to ensure that there is enough unvested balance. All chains using the `balances` module allow vesting, so this check makes sense.

Regarding the use of locks, `Locks` is a storage item of the entire module. If the logic makes it past the vesting check and the chain does not implement `Currency` or `LockableCurrency` then the `locks` getter will return empty and the function will return `Ok(())`.

```rust
let locks = Self::locks(who);
if locks.is_empty() {
	return Ok(())    // <--- Will return here
}
```

Reply:
I mean it is better to also expose `ensure_account_can_withdraw` in the `Currency` or `LockableCurrency` trait so other modules are able to check this liquidity before calling mutate method offered by `Currency` trait.